src/main/scala/freeeventsourcing/Event.scala

```diff
@@ -12,25 +13,29 @@ trait EventTime
   //TODO maybe a "trace" (cause of the event)
 }
 
+/** Selects a specific subset of events. */
 @typeclass trait EventSelector[S <: EventSelector.WithEventType] {
   def castEvent(e: Any): S#Event
-  def asTag(selector: S): EventTag //TODO Move into akka?
+
+  /** Topics are used for more efficient filtering of events. Only events on this topic will be delivered to the selector. */
+  def topic(selector: S): EventTopic
 }
 object EventSelector {
   type WithEventType = { type Event }
-
 }
 
-//TODO Move into akka
-case class EventTag(key: String, value: String)
+/** Topic for specific class of events. Ie all events from a certain aggregate type or for a specific aggregate. */
+case class EventTopic(composedTopic: CompositeName) extends AnyVal {
+  def topic = composedTopic.serialize
+}
 
 object EventTagger {
-  type EventTagger = Any ⇒ Traversable[EventTag] // TODO find different return type..
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Take the required action to fix the issue indicated by this comment.](https://www.codacy.com/app/msiegenthaler/free-event-sourcing/file/2612394303/issues/source?bid=3298199&fileBranchId=3311387#l33)